src/movies.py

```diff
@@ -0,0 +1,175 @@
+import requests
+import json
+
+
+def popular_movies(pages, headers):
+    """
+        This function calls the TMDB API for the popular movies list
+    
+        :param pages: int - number of pages of top rated movies to get there are 20 results on each page
+        :param headers: json - api authorization header
+        :return: json return the api response of popular movies
+    """
+    
+    # api endpoint
+    url = "https://api.themoviedb.org/3/movie/popular?language=en-US&page={}&region=840".format(pages) # region 840 is USA see https://en.wikipedia.org/wiki/ISO_3166-1 for other country codes
+
+    # use the api url and headers to get the information on the popular movies
+    response = requests.get(url, headers=headers)
+    return response.json()
+
+
+def toprated_movies(pages, headers):
+
+    """
+        This function calls the TMDB API for the top rated movies list
+    
+        :param pages: int - number of pages of top rated movies to get there are 20 results on each page
+        :param headers: json - api authorization header
+        :return: json return the api response of top rated movies
+    """
+    
+    # api endpoint
+    url = "https://api.themoviedb.org/3/movie/top_rated?language=en-US&page={}&region=840".format(pages)
+
+    # use the api url and headers to get the information on the popular movies
+    response = requests.get(url, headers=headers)
+    return response.json()
+
+
+def extract_movie_data(movies):
+    """
+        Take the movie id and title out of the API results
+
+        :param movies: json - a json of movie data
+        :return: json return only the extracted data from the movies should be just the ids and the title
+    """
+
+    # extract ID and Title
+    extracted_data = [
+        {"id": movie.get("id"), "title": movie.get("title")}
+        for movie in movies.get("results", [])
+        if 16 not in movie.get("genre_ids", []) # genre 16 is animation we are choosing to ignore it for now since it will be difficult to recognize actors
+    ]
+    return {"movies": extracted_data}
+
+
+def actors(movies, headers):
+    """
+        This function gets the actors from a given movie and then calls the popular_actors function to get the most popular ones
+
+        :param movies: json - a json of movie data
+        :param headers: json - api authorization header
+        :return: json return the movies json after adding the most popular actors to each movie
+    """
+    
+    # run an api call on each movie and get the cast, after that call popular_actors to get the most popular actors and add them to the json
+    for movie in movies.get("movies", []):
+        movie_id = movie.get("id")
+        # api endpoint
+        url = "https://api.themoviedb.org/3/movie/{}/credits?language=en-US".format(movie_id)
```

Comment:
It looks like `"https://api.themoviedb.org/3"` is a common base URL for a lot of requests we're making, could we put that in a constant and re-use it? (potentially in a `constants.py` file):
```suggestion
        url = "{}/movie/{}/credits?language=en-US".format(TMDB_BASE_URL, movie_id)
```
alternatively if that's a bit harder to read:
```suggestion
        url = f"{TMDB_BASE_URL}/movie/{movie_id}/credits?language=en-US"
```

Reply:
Does that constants file already exist I didn't see it. But yes that makes sense to me.

Reply:
Nope not yet!